docs/examples/self_attentive_sentence_embedding/self_attentive_sentence_embedding.md

```diff
@@ -0,0 +1,535 @@
+Each sample in the data is a user's comment, the language is English, and each comment category is marked 1-5, representing 5 different emotional colors.
+
+```{.python .input}
+# download the data zip from server.
+data_url = 'http://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/yelp_review_subset-167bb781.zip'
+zip_path = mx.gluon.utils.download(data_url)
+
+# unzip the zip file.
+zip_file = zipfile.ZipFile(zip_path)
+json_path = zip_file.extract(zip_file.namelist()[0])
+
+## load json data.
+with open(json_path, 'r', encoding='utf-8') as fr:
+    data = json.load(fr)
+
+# create a list of review a label paris.
+dataset = [[text, int(label)] for text, label in zip(data['texts'], data['labels'])]
+
+# randomly divide one percent from the training set as a verification set.
+train_dataset, valid_dataset = nlp.data.train_valid_split(dataset, 0.01)
+len(train_dataset), len(valid_dataset)
+```
+
+### Data Processing
+
+The purpose of the following code is to process the raw data so that the processed data can be used for model training and prediction. We will use the `SpacyTokenizer` to split the comments into strings, `ClipSequence` to crop the comments to the specified length, and build the dictionary based on the word frequency of the training data. Then we attach the [Glove](https://nlp.stanford.edu/pubs/glove.pdf)[2]  pre-trained word vector to the vocab and convert the characters of each comment data into the corresponding word index in the vocab.
+Finally get the standardized training data set and verification data set
+
+```{.python .input}
+# tokenizer takes as input a string and outputs a list of tokens.
+tokenizer = nlp.data.SpacyTokenizer('en')
+
+# length_clip takes as input a list and outputs a list with maximum length 100.
+length_clip = nlp.data.ClipSequence(100)
+
+def preprocess(x):
+
+    # Limit the range of labels to 0-4
+    data, label = x[0], x[1]-1
+
+    # clip the length of review words
+    data = length_clip(tokenizer(data))
+    return data, label
+
+def get_length(x):
+    return float(len(x[0]))
+
+def preprocess_dataset(dataset):
+    start = time.time()
+
+    with mp.Pool() as pool:
+        # Each sample is processed in an asynchronous manner.
+        dataset = gluon.data.SimpleDataset(pool.map(preprocess, dataset))
+        lengths = gluon.data.SimpleDataset(pool.map(get_length, dataset))
+    end = time.time()
+
+    print('Done! Tokenizing Time={:.2f}s, #Sentences={}'.format(end - start, len(dataset)))
+    return dataset, lengths
+
+# Preprocess the dataset
+train_dataset, train_data_lengths = preprocess_dataset(train_dataset)
+valid_dataset, valid_data_lengths = preprocess_dataset(valid_dataset)
+```
+
+```{.python .input}
+# create vocab
+train_seqs = [sample[0] for sample in train_dataset]
+counter = nlp.data.count_tokens(list(itertools.chain.from_iterable(train_seqs)))
+
+vocab = nlp.Vocab(counter, max_size=10000, padding_token=None, bos_token=None, eos_token=None)
```

Comment:
Yes, the standard way of writing is what you said.  because my data in this experiment is filled with 0, 0 corresponds to unk in the vocabulary, so padding_token is set to None.